### PR TITLE
Use clang-tidy to find performance issues

### PR DESCRIPTION
**Pull request by feliwir**
Use the [clang-tidy](https://clang.llvm.org/extra/clang-tidy) static code analyzer to find any performance issues within our source code. I've applied the following checks:
- [performance-faster-string-find](https://clang.llvm.org/extra/clang-tidy/checks/performance-faster-string-find.html): find any string find algorithms that can be improved
- [performance-for-range-copy](https://clang.llvm.org/extra/clang-tidy/checks/performance-for-range-copy.html): avoid unnecessary copies inside range based for loops
- [performance-inefficient-vector-operation](https://clang.llvm.org/extra/clang-tidy/checks/performance-inefficient-vector-operation.html): when growing a vector inside a loop, reserve the size before
- [performance-type-promotion-in-math-fn](https://clang.llvm.org/extra/clang-tidy/checks/performance-type-promotion-in-math-fn.html): avoid calling double variants of math functions if the input & output is a float
- [performance-unnecessary-copy-initialization](https://clang.llvm.org/extra/clang-tidy/checks/performance-unnecessary-copy-initialization.html): avoid creating any local variable which could be references

### CODE REVIEW

File: Source/Compat/processpool.cpp [hunk 1]
```diff
@@ -610,7 +610,8 @@ int ChildProcessMessage(const string& msg, const ProcessPool::JobMap &job_map){
         }
         int argc = (int)params_separated.size();
         vector<const char*> argv;
-        for(int i=0; i<argc; ++i){
+        argv.reserve(argc);
+for(int i=0; i<argc; ++i){
```
Comment (Conan-Kudo) [inline Source/Compat/processpool.cpp:614]: Broken indentation here.

File: Source/Scripting/angelscript/asfuncs.cpp [hunk 2]
```diff
@@ -2240,7 +2240,8 @@ int ASDebugDrawLines( const CScriptArray &array, vec4 color, int lifespan_int )
 
     std::vector<vec3> data;
 
-    for( int n = 0; n < (int)array.GetSize(); n++ )
+    data.reserve((int)array.GetSize());
+for( int n = 0; n < (int)array.GetSize(); n++ )
```
Comment (Conan-Kudo) [inline Source/Scripting/angelscript/asfuncs.cpp:2244]: Broken indentation.

File: Source/Scripting/angelscript/asfuncs.cpp [hunk 3]
```diff
@@ -3941,7 +3942,8 @@ bool AS_ImGui_Combo(const std::string& label, int& current_item, const CScriptAr
     const int items_count = items.GetSize();
     std::vector<const char*> items_data;
 
-    for(int n = 0; n < items_count; ++n) {
+    items_data.reserve(items_count);
+for(int n = 0; n < items_count; ++n) {
```
Comment (Conan-Kudo) [inline Source/Scripting/angelscript/asfuncs.cpp:3946]: Broken indentation.

File: Source/Scripting/angelscript/asfuncs.cpp [hunk 4]
```diff
@@ -4257,7 +4259,8 @@ bool AS_ImGui_ListBox(const std::string& label, int& current_item, const CScript
...
```
Comment (Conan-Kudo) [inline Source/Scripting/angelscript/asfuncs.cpp:4263]: Broken indentation.

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

**Comment by feliwir** [comment]
All review comments adressed & rebased

**Review by Conan-Kudo — APPROVED** [review]

File: Source/Scripting/angelscript/add_on/scripthelper/scripthelper.cpp [hunk 5]
```diff
@@ -228,7 +228,7 @@ int WriteConfigToStream(asIScriptEngine *engine, ostream &strm)
 			for(;;)
 			{
 				// Find " characters
-				pos = str.find("\"",pos);
+				pos = str.find('\"',pos);
```
Comment (shinymerlyn) [inline Source/Scripting/angelscript/add_on/scripthelper/scripthelper.cpp:231]: this code is inside angelscript I think? I believe these local patches may make it harder to diff when upgrading the angelscript version.

Comment (shinymerlyn) [reply Source/Scripting/angelscript/add_on/scripthelper/scripthelper.cpp:231]: or maybe this is a local "add-on". not sure if it's one of the standard ones or not.

**Pull request merged**